src/snowflake/snowpark/_internal/server_connection.py

```diff
@@ -171,6 +173,22 @@ def __init__(
             except TypeError:
                 pass
 
+        # thread safe param protection
+        self._thread_safe_session_enabled = self._lower_case_parameters.get(
+            "python_snowpark_enable_thread_safe_session",
+            self._get_client_side_session_parameter(
+                "PYTHON_SNOWPARK_ENABLE_THREAD_SAFE_SESSION", False
+            ),
+        )
```

Comment:
If connection parameter sets `PYTHON_SNOWPARK_ENABLE_THREAD_SAFE_SESSION`, we read the value, otherwise we read the value from session.

Reply:
i think i see where the problem is now, we are also making the ServerConnection object thread safe, and the parameter here is a connection level parameter not session level parameter, that means customer will not be able to turn this on and off through the session object, they will have to turn it on an off through connection, and in general starting a new connection/session is required to turn this off.

Reply:
is setting through connection parameter a documented way for user to turn things on and off?

Reply:
and is _get_client_side_session_parameter also going to take care of the connection parameter, from the test, i believe it should be taken care also

Reply:
I documented in changelog.

I added this parameter mainly for testing since the locks cannot be changed during session.

Reply:
remove changelog as per offline discussion